### PR TITLE
Hw_task_2

### PR DESCRIPTION
**Pull request by AlenaDrevina**
Task2 was done and readmi was re-written.

### CODE REVIEW

**Comment by AlenaDrevina** [comment]
push to main

File: HW_module_2.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import string
+import random
+
+# create a list of random number of dicts (from 2 to 10)
+
+dict_List = []                                             # empty list
+for x in range(random.randint(2, 10)):                     # random number of dictionaries
+#    size = random.randint(1,26)                          # random dictionary size
```
Comment (AntonLazarchik) [inline HW_module_2.py:8]: Why have you chosen size 3 in the final version? size = random.randint(1,26) looks ok.
If you wanted to expand the length you could use size = random.randint(5,26) for example.

File: HW_module_2.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import string
+import random
+
+# create a list of random number of dicts (from 2 to 10)
+
+dict_List = []                                             # empty list
```
Comment (AntonLazarchik) [inline HW_module_2.py:6]: dict_list

File: HW_module_2.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+for x in range(random.randint(2, 10)):                     # random number of dictionaries
+#    size = random.randint(1,26)                          # random dictionary size
+    size = 3                                               # the size of dictionary
+    keys = random.sample(string.ascii_lowercase, size)     # random lowercase letters in key parameters
+    values = (random.randint(1, 50) for y in range(size))  # random numbers in value parameters
```
Comment (AntonLazarchik) [inline HW_module_2.py:11]: Why do you need the generator here?  values = random.sample(range(1, 100), size)
In some cases, a generator can be a better option, but I'm not sure that it was used here intentionally.

File: HW_module_2.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    one_Dict = dict(zip(keys, values))                      #the dictionary can be created by pairing up the letters
```
Comment (AntonLazarchik) [inline HW_module_2.py:12]: one_dict

File: HW_module_2.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    dict_List.append(one_Dict)                              # add it to the list
+
+print(dict_List)                                           # view the result
+
+# get previously generated list of dicts and create one common dict
+
+distinct_keys = []                          # create an empty list of distinct letters from all dictionaries in our list
+duplicated_keys = []                        # create an empty list of distinct letters which have duplicates
+
+for z in dict_List:                         # go through each dictionary
```
Comment (AntonLazarchik) [inline HW_module_2.py:23]: please use meaningful names. instead of a, z, i and so on.

File: HW_module_2.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    if key in duplicated_keys:              # if this letter is in "duplicated_keys" list
+        max_value = 0                       # then let's define max value for each key (the default is zero)
+        dict_number = 0                     # define the number of the dictionary in the list
+        for i, rand_dict in enumerate(dict_List):  # iterate over each dictionary and its index from the list using the function enumerate()
+            if key in rand_dict and rand_dict[key] > max_value:  # check if the specified key is in the specified dictionary
+                                                                 # AND the key's value is bigger than max value
+                max_value = rand_dict[key]  # then update max value
+                dict_number = i + 1         # then update the number of the dictionary (started by 1)
+        key_final = key + '_' + str(dict_number)  # create the key
+        one_common_dict.setdefault(key_final, max_value)  # insert key and its max value in the one_common_dict
```
Comment (AntonLazarchik) [inline HW_module_2.py:48]: setdefault works only for the initial insert of the key with an optional default value. Insert key with a value of default will be performed only if the key is not in the dictionary.
In this case the common option to insert a new key into dict can be used:
one_common_dict[key_final] = max_value

**Pull request merged**